projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -1,20 +1,24 @@
 import { InnerBlocks, store as blockEditorStore } from '@wordpress/block-editor';
-import { createBlock, getBlockType, registerBlockVariation } from '@wordpress/blocks';
+import { cloneBlock, createBlock, getBlockType, registerBlockVariation } from '@wordpress/blocks';
 import { Placeholder } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { get } from 'lodash';
 import PaymentsIntroBlockPicker from './block-picker';
+import PaymentsIntroPatternPicker from './pattern-picker';
 import defaultVariations from './variations';
 
 export default function JetpackPaymentsIntroEdit( { name, clientId, className } ) {
-	const { blockType, hasInnerBlocks } = useSelect( select => {
-		const { getBlocks } = select( blockEditorStore );
+	const patternFilter = pattern => pattern.categories.includes( 'earn' );
+
+	const { blockType, hasInnerBlocks, hasPatterns } = useSelect( select => {
+		const { getBlocks, __experimentalGetAllowedPatterns } = select( blockEditorStore );
 
 		return {
 			blockType: getBlockType( name ),
 			hasInnerBlocks: getBlocks( clientId )?.length > 0,
+			hasPatterns: __experimentalGetAllowedPatterns().filter( patternFilter ).length > 0,
```

Comment:
Let's add a safeguard here in case the experimental selector is removed.
```suggestion
			hasPatterns: __experimentalGetAllowedPatterns?.().filter( patternFilter ).length > 0,
```